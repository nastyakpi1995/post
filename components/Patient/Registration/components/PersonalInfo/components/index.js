import Contact from './Contact';
import General from './General';
import Personal from './Personal';

export { Contact, General, Personal };
