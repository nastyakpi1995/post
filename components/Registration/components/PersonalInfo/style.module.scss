.wrapper {
  display: block;
  background: #252f4d;
  min-height: 100vh;

  .container {
    background: #f5fafe;
    max-width: 1074px;
    border-radius: 25px 25px 0 0;
    margin: 0 auto;

    .validation {
      display: block;
      width: 40%;
      max-width: 330px;
      margin: 0 auto;
      height: 100%;

      .validation-title {
        padding-top: 70px;
        margin: 0;
        font-family: Avenir Next;
        font-style: normal;
        font-weight: 600;
        color: #252f4d;
        font-size: 24px;
      }

      .validation-button {
        padding: 30px 0;

        .validation-button__saveProfile {
          width: 100%;
          text-transform: uppercase;
          display: block;
          color: #fff;
          border: 1px solid #0fc7c7;
          outline: none;
          cursor: pointer;
          height: 48px;
          background: #0fc7c7;
          border-radius: 24px;
          font-family: Avenir Next;
          font-style: normal;
          font-weight: 600;
          font-size: 14px;
          text-align: center;
          letter-spacing: 1px;
        }
      }
    }
  }
}

@media (max-width: 980px) {
  .validation {
    width: 100%;
  }
  .container {
    max-width: 95% !important;
  }
}

@media (min-height: 680px) {
  .container {
    height: 100vh !important;
  }
}