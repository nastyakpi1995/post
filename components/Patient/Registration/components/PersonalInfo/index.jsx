import React from 'react';
import { withTranslation } from 'i18n';
import { Formik } from 'formik';
import Header from '../../../../view/objects/AuthHeader';
import { Button } from '../../../../view/ui';
import { General, Contact, Personal } from './components';
import { GENDER_TYPES, COUNTRIES, CITIES } from '../../constants';
import styles from './style.module.scss';
import RegisterSchema from '../../Forms/registerForm/ConfigmationForm';

function PersonalInfo({ onHandleChangePage = () => {}, t }) {
  return (
    <div className={styles.wrapper}>
      <Header />
      <div className={styles.container}>
        <div className={styles.validation}>
          <h1 className={styles['validation-title']}>
            {t('personalInfo.title')}
          </h1>
          <Formik
            initialValues={{
              firstName: '',
              lastName: '',
              nationalId: '',
              dateOfBirth: '2002-08-16',
              gender: GENDER_TYPES[0],
              country: COUNTRIES[0],
              city: CITIES[0],
              direction: '',
              houseNumber: '',
              apartmentNumber: '',
              floorNumber: '',
              zipCode: '',
              email: '',
            }}
            validationSchema={RegisterSchema(t)}
            onSubmit={(values, actions) => {
              // TODO: should save values to redux here
              onHandleChangePage();
            }}
          >
            {({
              values,
              errors,
              touched,
              handleChange,
              handleBlur,
              handleSubmit,
            }) => (
              <form onSubmit={handleSubmit}>
                <General
                  values={values}
                  errors={errors}
                  touched={touched}
                  handleChange={handleChange}
                  handleBlur={handleBlur}
                />
                <Personal
                  values={values}
                  errors={errors}
                  touched={touched}
                  handleChange={handleChange}
                  handleBlur={handleBlur}
                />
                <Contact
                  values={values}
                  errors={errors}
                  touched={touched}
                  handleChange={handleChange}
                  handleBlur={handleBlur}
                />
                <div className={styles['validation-button']}>
                  <Button
                    type="submit"
                    className={styles['validation-button__saveProfile']}
                    text={t('personalInfo.save')}
                  />
                </div>
              </form>
            )}
          </Formik>
        </div>
      </div>
    </div>
  );
}

export default withTranslation('registration')(PersonalInfo);
