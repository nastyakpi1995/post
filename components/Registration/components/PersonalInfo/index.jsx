import React, { useEffect } from 'react';
import { withTranslation } from 'i18n';
import { Formik, Form } from 'formik';
import { connect } from 'react-redux';
import { getCountries, getCities } from 'countries-cities';
import PropTypes from 'prop-types';
import Header from '../../../view/objects/AuthHeader';
import { Button } from '../../../view/ui';
import { General, Contact, Personal } from './components';
import { GENDER_TYPES } from '../../constants';
import styles from './style.module.scss';
import RegisterPersonalSchema from '../../Forms/registerForm/PersonalForm';
import * as registrationActions from '../../../../redux/actions/registrationActions';

function PersonalInfo({
  onHandleChangePage,
  t,
  personalInfoSuccess,
  onRequestPersonalInfo,
  personalInfoErrors,
  onResetErrors,
  type,
}) {
  useEffect(() => {
    if (personalInfoSuccess) {
      onHandleChangePage();
    }
  }, [personalInfoSuccess]);

  return (
    <div className={styles.wrapper}>
      <Header />
      <div className={styles.container}>
        <div className={styles.validation}>
          <h1 className={styles['validation-title']}>
            {t('personalInfo.title')}
          </h1>
          <Formik
            initialValues={{
              firstName: '',
              lastName: '',
              nationalId: '',
              dateOfBirth: '2002-08-16',
              gender: GENDER_TYPES[0],
              country: getCountries()[21], // Belgium
              city: getCities(getCountries()[21])[0],
              direction: '',
              houseNumber: '',
              apartmentNumber: '',
              floor: '',
              zipCode: '',
              email: '',
            }}
            validationSchema={RegisterPersonalSchema(t)}
            onSubmit={(values, actions) => {
              setTimeout(() => {
                const updatedData = {
                  ...values,
                  gender: values.gender === GENDER_TYPES[0] ? 'm' : 'fm',
                };
                onRequestPersonalInfo(updatedData);
              }, 0);
            }}
          >
            {({
              values,
              errors,
              touched,
              handleChange,
              handleBlur,
              handleSubmit,
              setFieldValue,
            }) => (
              <Form onSubmit={handleSubmit}>
                <General
                  values={values}
                  errors={errors}
                  touched={touched}
                  handleChange={handleChange}
                  handleBlur={handleBlur}
                  personalInfoErrors={personalInfoErrors}
                  onResetErrors={onResetErrors}
                />
                <Personal
                  values={values}
                  errors={errors}
                  touched={touched}
                  handleChange={handleChange}
                  handleBlur={handleBlur}
                  setFieldValue={setFieldValue}
                  personalInfoErrors={personalInfoErrors}
                  onResetErrors={onResetErrors}
                />
                <Contact
                  values={values}
                  errors={errors}
                  touched={touched}
                  handleChange={handleChange}
                  handleBlur={handleBlur}
                  setFieldValue={setFieldValue}
                  personalInfoErrors={personalInfoErrors}
                  onResetErrors={onResetErrors}
                />
                <div className={styles['validation-button']}>
                  <Button
                    type="submit"
                    className={styles['validation-button__saveProfile']}
                    text={t('personalInfo.save')}
                  />
                </div>
              </Form>
            )}
          </Formik>
        </div>
      </div>
    </div>
  );
}

PersonalInfo.propTypes = {
  onHandleChangePage: PropTypes.func,
  t: PropTypes.func,
  onRequestSignUp: PropTypes.func,
  serverErrors: PropTypes.object,
  generalSuccess: PropTypes.bool,
  type: PropTypes.string,
  onResetErrors: PropTypes.func,
};

const mapDispatchToProps = (dispatch) => ({
  onRequestPersonalInfo: (data) =>
    dispatch(registrationActions.requestPersonalInfo(data)),
  onResetErrors: (data) =>
    dispatch(registrationActions.errorPersonalInfo(data)),
});

const mapStateToProps = (state) => ({
  personalInfo: state.registration.personalInfo,
  personalInfoErrors: state.registration.personalInfoErrors,
  personalInfoSuccess: state.registration.personalInfoSuccess,
});

export default withTranslation('registration')(
  connect(mapStateToProps, mapDispatchToProps)(PersonalInfo),
);
