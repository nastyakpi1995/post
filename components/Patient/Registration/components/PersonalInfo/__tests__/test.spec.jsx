import React from 'react';
import renderer from 'react-test-renderer';
import PersonalInfo from '../index';

it('renders correctly', () => {
  const component = renderer.create(<PersonalInfo />).toJSON();
  expect(component).toMatchSnapshot();
});
