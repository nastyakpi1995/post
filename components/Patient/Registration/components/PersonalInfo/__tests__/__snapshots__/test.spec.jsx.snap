// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <div>
      <a
        className="
          function link() { [native code] }
          undefined
        "
        href="#"
        id="mainPage"
      >
        <span>
          Doc
        </span>
        <span>
          Woo
        </span>
      </a>
    </div>
    <div>
      <button
        id="doctorLogin"
        type="button"
      >
        login as doctor
      </button>
      <select
        id="language"
      >
        <option>
          en
        </option>
        <option>
          ua
        </option>
      </select>
    </div>
  </div>
  <div>
    <div>
      <h1>
        Complete Your Profile
      </h1>
      <div>
        <p>
          General information
        </p>
        <div>
          <label>
            first name
            <div
              className="undefined"
            >
              <input
                id="firstName"
                name="firstName"
                placeholder="Frederick"
                type="text"
              />
            </div>
          </label>
        </div>
        <div>
          <label>
            last name
            <div
              className="undefined"
            >
              <input
                id="lastName"
                name="lastName"
                placeholder="Valdez"
                type="text"
              />
            </div>
          </label>
        </div>
        <div>
          <label>
            national id
            <div
              className="undefined"
            >
              <input
                id="nationalId"
                name="nationalId"
                placeholder="0123456789"
                type="number"
              />
            </div>
          </label>
        </div>
      </div>
      <div>
        <p>
          Personal information
        </p>
        <div>
          <div>
            <label>
              date of birth
              <div
                className="undefined"
              >
                <input
                  id="date"
                  name="date"
                  placeholder="2017-06-01"
                  type="date"
                />
              </div>
            </label>
          </div>
          <div>
            <label>
              gender
              <select
                id="gender"
              >
                <option>
                  Male
                </option>
                <option>
                  Female
                </option>
              </select>
            </label>
          </div>
        </div>
      </div>
      <div>
        <p>
          Contact information
        </p>
        <div>
          <div>
            <label>
              country
              <select
                id="country"
              >
                <option>
                  Belgium
                </option>
                <option>
                  Ukraine
                </option>
              </select>
            </label>
          </div>
          <div>
            <label>
              city
              <select
                id="city"
              >
                <option>
                  Brussels
                </option>
                <option>
                  Kyiv
                </option>
              </select>
            </label>
          </div>
        </div>
        <div>
          <label>
            direction
            <div
              className="undefined"
            >
              <input
                id="direction"
                name="direction"
                placeholder="Rue Montagne aux Herbes PotagÃ¨res"
                type="text"
              />
            </div>
          </label>
        </div>
        <div>
          <div>
            <label>
              house number
              <div
                className="undefined"
              >
                <input
                  id="house"
                  name="house"
                  placeholder="26"
                  type="number"
                />
              </div>
            </label>
          </div>
          <div>
            <label>
              apartment number
              <div
                className="undefined"
              >
                <input
                  id="apartment"
                  name="apartment"
                  placeholder="3"
                  type="number"
                />
              </div>
            </label>
          </div>
          <div>
            <label>
              floor (optional)
              <div
                className="undefined"
              >
                <input
                  id="floor"
                  name="floor"
                  placeholder="2"
                  type="number"
                />
              </div>
            </label>
          </div>
        </div>
        <div>
          <label>
            zip code
            <div
              className="undefined"
            >
              <input
                id="zipCode"
                name="zipCode"
                placeholder="1000"
                type="number"
              />
            </div>
          </label>
        </div>
        <div>
          <label>
            email
            <div
              className="undefined"
            >
              <input
                id="email"
                name="email"
                placeholder="rowe.shana@mina.biz"
                type="text"
              />
            </div>
          </label>
        </div>
      </div>
      <div>
        <button
          id="date"
          type="button"
        >
          save profile data
        </button>
      </div>
    </div>
  </div>
</div>
`;
